### DISCUSSION TITLE
How to send additional information via HttpData?

### DISCUSSION THREAD

**Discussion by hlange** [eclipse-edc/Connector#1386] [Q&A]
Is it possible to send additional info to the provider?

Currently, I send the following request:
```
{
  "asset": {
    "properties": {
      "asset:prop:id": "asset-1",
      "asset:prop:name": "asset name",
      "asset:prop:contenttype": "aapplication/json",
      "asset:prop:policy-id": "use-eu"
    }
  },
  "dataAddress": {
    "properties": {
      "endpoint": "http://localhost:8000",
      "name":"",
      "type": "HttpData"
    }
  }
}
```
The idea is that I also send to the provider additional info like a filter or a yaml file for docker-compose.

**Comment by jimmarino** [comment]
Can you describe your use case and what request you are sending to the *provider*?

**Reply by hlange** [reply]
An idea would be to send a yaml file or a configuration to process the data before sending it to the consumer,  as a very simple case of code2data.

**Reply by jimmarino** [reply]
This can be done using the extensible properties on `TransferRequestDto` which will be placed on `DataRequest`. You can then write a custom `ResourceManifestGenerator` and `Provisioner` to run the data processing step where it will have access to the configuration you send.

The generator and provisioner can either be written in Java using the SPI or an HTTP endpoint using the HTTP provisioner.

**Reply by hlange** [reply]
I need a bit of help, it makes sense what you are saying, but I have no clue where I need to start.

1. How can I extend the TransferRequestDto  use them with the DataRequest?
2. My first Idea would be to use the Http provisoner which would reduce that I need to make changes in the source code, but it is not documented, so I don't know where to start.
3. I think it would be great if you can define in the future with the datamanagment api at the creation of an asset if additional parameter are necessary which should be forwarded to the backend service or something like this.

Sorry if the answer is obvious, but currently I can't see it

**Reply by jimmarino** [reply]
1. You don't need to extend TransferRequestDto - set the data you want as properties.
2. You shouldn't need to make any changes in the source code. Implement a provisioner extension that has a `ProviderResourceDefinitionGenerator` which reads the properties from DataRequest (they will be copied from the TransferRequestDto by the runtime) and creates a ResourceDefinition used for provisioning. Looking into this further, the HTTP provisioner currently does not support passing extensible properties to a provisioning endpoint but could be added in the future. This would need to be looked into, so an extension is your best short-term bet.
3. Extensible properties are designed to do this. There is a separate but related issue of how to describe the schema of additional data needed for making a transfer request.

**Reply by dilip-dhankecha** [reply]
@jimmarino @hlange 
Do you guys having any example or sample code for the same?
